textinput_model.go

```diff
@@ -30,114 +30,131 @@ func (p *textinputPlugin) addChar(char []rune) {
 
 	p.selectionBase += len(char)
 	p.selectionExtent = p.selectionBase
-	p.updateEditingState()
 }
 
-func (p *textinputPlugin) MoveCursorHome(modsIsModifier bool, modsIsShift bool, modsIsWordModifierShift bool, modsIsWordModifier bool) {
+func (p *textinputPlugin) MoveCursorHomeSimple() {
+	p.selectionBase = 0
+	p.selectionExtent = p.selectionBase
+}
+
+func (p *textinputPlugin) MoveCursorHomeSelect() {
 	p.selectionBase = 0
-	if !modsIsShift {
-		p.selectionExtent = p.selectionBase
-	}
-	p.updateEditingState()
 }
 
-func (p *textinputPlugin) MoveCursorEnd(modsIsModifier bool, modsIsShift bool, modsIsWordModifierShift bool, modsIsWordModifier bool) {
+func (p *textinputPlugin) MoveCursorEndSimple() {
 	p.selectionBase = len(p.word)
-	if !modsIsShift {
-		p.selectionExtent = p.selectionBase
+	p.selectionExtent = p.selectionBase
+}
+
+func (p *textinputPlugin) MoveCursorEndSelect() {
+	p.selectionBase = len(p.word)
+}
+
+func (p *textinputPlugin) MoveCursorLeftSimple() {
+	p.selectionExtent--
+}
+
+func (p *textinputPlugin) MoveCursorLeftWord() {
+	p.selectionBase = indexStartLeadingWord(p.word, p.selectionBase)
+	p.selectionExtent = p.selectionBase
+
+}
+
+func (p *textinputPlugin) MoveCursorLeftLine() {
+	if p.isSelected() {
+		p.selectionExtent = indexStartLeadingWord(p.word, p.selectionExtent)
+	} else {
+		p.selectionExtent = indexStartLeadingWord(p.word, p.selectionBase)
 	}
-	p.updateEditingState()
+
 }
 
-func (p *textinputPlugin) MoveCursorLeft(modsIsModifier bool, modsIsShift bool, modsIsWordModifierShift bool, modsIsWordModifier bool) {
-	if modsIsWordModifierShift {
-		if p.isSelected() {
-			p.selectionExtent = indexStartLeadingWord(p.word, p.selectionExtent)
-		} else {
-			p.selectionExtent = indexStartLeadingWord(p.word, p.selectionBase)
-		}
-	} else if modsIsWordModifier {
-		p.selectionBase = indexStartLeadingWord(p.word, p.selectionBase)
-		p.selectionExtent = p.selectionBase
-	} else if modsIsShift {
-		p.selectionExtent--
-	} else if !p.isSelected() {
+func (p *textinputPlugin) MoveCursorLeftReset() {
+	if !p.isSelected() {
 		if p.selectionBase > 0 {
 			p.selectionBase--
 			p.selectionExtent = p.selectionBase
 		}
 	} else {
 		p.selectionBase = p.selectionExtent
 	}
-	p.updateEditingState()
 }
 
-func (p *textinputPlugin) MoveCursorRight(modsIsModifier bool, modsIsShift bool, modsIsWordModifierShift bool, modsIsWordModifier bool) {
-	if modsIsWordModifierShift {
-		if p.isSelected() {
-			p.selectionExtent = indexEndForwardWord(p.word, p.selectionExtent)
-		} else {
-			p.selectionExtent = indexEndForwardWord(p.word, p.selectionBase)
-		}
-	} else if modsIsWordModifier {
-		p.selectionBase = indexEndForwardWord(p.word, p.selectionBase)
-		p.selectionExtent = p.selectionBase
-	} else if modsIsShift {
-		p.selectionExtent++
-	} else if !p.isSelected() {
+func (p *textinputPlugin) MoveCursorRightSimple() {
```

Comment:
The `Simple` is misleading.
How about: `s/MoveCursorRightSimple/extentSelectionRight()`